renku/core/management/datasets.py

```diff
@@ -383,12 +385,29 @@ def set_dataset_images(self, dataset, images, safe_image_paths=[]):
             if existing:
                 dataset.images.append(existing)
                 continue
-
+            image_type = None
             if urlparse(content_url).netloc:
                 # NOTE: absolute url
-                dataset.images.append(ImageObject(content_url, position, id=ImageObject.generate_id(dataset, position)))
-                images_updated = True
-                continue
+                if not img.get("mirror_locally", False):
+                    dataset.images.append(
+                        ImageObject(content_url, position, id=ImageObject.generate_id(dataset, position))
+                    )
+                    images_updated = True
+                    continue
+
+                # NOTE: mirror the image locally
+                try:
+                    path, _ = urlretrieve(content_url)
```

Comment:
Not really any reason, other than it being shorter/a bit more readable than:
```
r = requests.get(url, stream=True)
if r.status_code == 200:
    path = tmpdir.mkdir() / create_temp_filename_somehow()
    with open(path, 'wb') as f:
        r.raw.decode_content = True
        shutil.copyfileobj(r.raw, f)       
```
or
```
r = requests.get(url, stream=True)
if r.status_code == 200:
    path = tmpdir.mkdir() / create_temp_filename_somehow()
    with open(path, 'wb') as f:
        for chunk in r:
            f.write(chunk)
```

We don't have a generic download method that uses requests anywhere, to my knowledge. `urlretrieve` just deals with quite a bit of logic that we'd have to otherwise implement ourselves.